### ISSUE TITLE
Add eslint-friendly-formatter

### ISSUE THREAD

**Issue by monolithed** [eslint/eslint#6648]
[eslint-friendly-formatter](https://www.npmjs.com/package/eslint-friendly-formatter) is one the best reporters:

![img-2016-06-25-14-44-20](https://cloud.githubusercontent.com/assets/803674/16356523/79ccaba8-3ae3-11e6-8f5a-c04d210df1cc.png)
![img-2016-06-25-14-47-01](https://cloud.githubusercontent.com/assets/803674/16356532/dd76c03a-3ae3-11e6-93e3-c16260a28db9.png)
![img-2016-06-25-14-48-05](https://cloud.githubusercontent.com/assets/803674/16356533/dd76cc6a-3ae3-11e6-8108-f7ec6e9a4c06.png)
![img-2016-06-25-14-46-29](https://cloud.githubusercontent.com/assets/803674/16356534/dd76d2be-3ae3-11e6-8f46-445477fda22f.png)
![clickable-rules2](https://cloud.githubusercontent.com/assets/803674/16715459/e52ded56-46e8-11e6-8869-cbe36b7cb53a.png)

This module is based on the original stylish formatter that is now part of ESLint, it adds the following:
— All the errors are reported at the end, so no more search for errors between tons of report lines
— Rule are clickable on terminals like iTerm2 (or standard Terminal in Mac OS) and take you to the rule documentation site.
— A summary is shown at the end with the offended rule id, rules are also clickable.
— It also shows a bit of context where the error happened, making easier to understand the nature of the error

Can't understand why is not standard formatter...

```
"lint": "eslint . --cache --format node_modules/eslint-friendly-formatter --ext .js",
```

That's not very user-friendly for this formater

**Comment by platinumazure** [comment]
Per http://eslint.org/docs/maintainer-guide/issues.html#types-of-issues, since this is a new formatter (to us), I have marked this as a feature request instead of an enhancement.

**Comment by nzakas** [comment]
We aren't adding any more formatters to the core. Since this formatter already exists, it would be best to remain as a separate package.
